### PR TITLE
#1358 Sensor data types

### LINKED ISSUE #1358: Add Sensor & LocationType
## Is your feature request related to a problem? Please describe.
Need to add the listed data types for the vaccine module

## Describe the solution you'd like

Add `Sensor` and `LocationType` and `BreachTemperatureConfiguration` as well as related fields **Note: Not `Location`**:

image

image

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
#### NOTE: Branched from #feature-vaccines

Fixes #1358 

## Change summary

- Adds `SensorLog` `SensorLogItemBatchJoin` and `Sensor` classes

- I'm second guessing having the BLE interfacing within the `Sensor` class
- Keeping `SensorLog` as a class,  I think a couple of methods might be able to go in here to clean more things up.

Comment here in the epic regarding these potential issues https://github.com/openmsupply/mobile/issues/1353#issuecomment-544221296

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

**Comment by joshxg** [comment]
We hadn't done previously due to realm being out of date. I think we should use `linkingObjects` though.

**Comment by wlthomson** [comment]
> We hadn't done previously due to realm being out of date. I think we should use `linkingObjects` though. Although I'm not 100% on hwo that works with migration. I'm gonna have to test it out

I think vaccines already going to be complicated and touch a lot of areas of code for one branch. I'd recommend doing this at a later time on a separate branch. If we end up replicating the amount of tests we had for data table for vaccines, it could get pretty messy. I think we should probably kick everything non-vital to a follow up after core vaccines is merged.

I'd suggest merging the basic messy vaccines first, test that (then master has vaccines and we relieve any pressure on that front), then a follow up for tidying it up, test that.

File: src/database/DataTypes/Sensor.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    try {
+      const result = await BleManager.getDevices(manufacturerId, this.macAddress);
+
+      const { success, data } = result;
+
+      if (!success || !data || data.length === 0) return result;
+
+      const advertisement = data[0];
+
+      const { macAddress, batteryLevel, temperature, logInterval } = advertisement;
```
Comment (wlthomson) [inline src/database/DataTypes/Sensor.js]: If going with destructuring, could even do:

```
const [ { macAddress, batteryLevel, temperature, logInterval } ] = data;
```

**Comment by joshxg** [comment]
Closing this due to other priorities. Not deleting branch and will reopen and address concerns at a later date. Sorry and thanks @wlthomson

**Pull request closed without merging**